backend/routes/quests.go

```diff
@@ -103,3 +118,100 @@ func GetTodayStartTime(w http.ResponseWriter, r *http.Request) {
 
 	WriteDataJson(w, "\""+string(todayStartTime.UTC().Format(time.RFC3339))+"\"")
 }
+
+func SendChat(w http.ResponseWriter, r *http.Request) {
+	userAddress := r.URL.Query().Get("address")
+	if userAddress == "" {
+		WriteErrorJson(w, http.StatusBadRequest, "Missing address parameter")
+		return
+	}
+
+	var chat FactionChat
+	err := json.NewDecoder(r.Body).Decode(&chat)
+	if err != nil {
+		WriteErrorJson(w, http.StatusBadRequest, "Invalid request body")
+		return
+	}
+	defer r.Body.Close()
+
+	err = core.PostgresExec(r.Context(), "INSERT INTO FactionChats (sender, faction_key, message) VALUES ($1, $2, $3)", chat.Sender, chat.FactionKey, chat.Message)
+	if err != nil {
+		WriteErrorJson(w, http.StatusInternalServerError, "Failed to send chat")
+		return
+	}
+
+	w.WriteHeader(http.StatusCreated)
+}
+
+func DeleteChat(w http.ResponseWriter, r *http.Request) {
+	userAddress := r.URL.Query().Get("address")
+	if userAddress == "" {
+		WriteErrorJson(w, http.StatusBadRequest, "Missing address parameter")
+		return
+	}
+
+	chatID := r.URL.Query().Get("id")
+	if chatID == "" {
+		WriteErrorJson(w, http.StatusBadRequest, "Missing chat ID parameter")
+		return
+	}
+
+	canDelete := false
+
+	leader, err := core.PostgresQueryOne[string]("SELECT ftns.leader FROM Factions ftns INNER JOIN FactionChats ftcs ON ftns.faction_key = ftcs.key WHERE ftcs.id = $1", chatID)
```

Comment:
Please compress all of these back and forth checks into a single query.

You can use a hardcoded 0 address for the user here too for now.